Sources/Clients/ProfileStore/ProfileStore.swift

```diff
@@ -611,15 +613,24 @@ extension ProfileStore {
 	static func createDeviceInfo() async throws -> ProfileSnapshot.Header.UsedDeviceInfo {
 		@Dependency(\.date) var dateGenerator
 		@Dependency(\.device) var device
+		@Dependency(\.uuid) var uuid
 		@Dependency(\.secureStorageClient) var secureStorageClient
 
 		let date = dateGenerator.now
 
+		let deviceIdentifier: UUID
+
+		if let existing = try? await secureStorageClient.loadDeviceIdentifier() {
+			deviceIdentifier = existing
+		} else {
+			deviceIdentifier = uuid()
```

Comment:
If we failed to load the `deviceIdentifier`, generate new one, but do NOT save it (yet).